### PR TITLE
Update includes to support foreign key relations

### PR DESCRIPTION
**Pull request by dgorelik**
This fleshes out https://github.com/google/python-spanner-orm/pull/87 to add initial support for foreign keys.

### CODE REVIEW

**Review by akefeli — APPROVED** [review]

File: spanner_orm/admin/update.py [hunk 1]
```diff
@@ -342,7 +334,7 @@ def validate(self) -> None:
     if db_index.primary:
       raise error.SpannerError('Index {} is the primary index'.format(
           self._index))
-
+
```
Comment (dgorelik) [inline spanner_orm/admin/update.py:337]: TODO: clean up here.

File: spanner_orm/condition.py [hunk 2]
```diff
@@ -406,11 +407,38 @@ def _validate(self, model_class: Type[Any]) -> None:
...
+    Args:
+      relation: Name of the relationship on the origin model or the Relationship/
+        ForeignKeyRelationship on the origin model class used to retrieve
+        associated objects
+      conditions: Conditions to apply on the subquery
+      foreign_key_relation: True if the relation is a foreign key relation,
+       False if it is a legacy relation (eg not enforced in Spanner)
+    """
     super().__init__()
+    self.foreign_key_relation = foreign_key_relation
```
Comment (dgorelik) [inline spanner_orm/condition.py:430]: Can we simply infer based on type here?

File: spanner_orm/condition.py [hunk 3]
```diff
@@ -429,21 +460,31 @@ def conditions(self) -> List[Condition]:
...
           'Condition must be bound before conditions is called')
     relation_conditions = []
-    for constraint in self.relation.constraints:
-      # This is backward from what you might imagine because the condition will
-      # be processed from the context of the destination model
-      relation_conditions.append(
-          ColumnsEqualCondition(constraint.destination_column,
-                                constraint.origin_class,
-                                constraint.origin_column))
+    if not self.foreign_key_relation:
```
Comment (dgorelik) [inline spanner_orm/condition.py:463]: Switch order to match below.

File: spanner_orm/foreign_key_relationship.py [hunk 4]
```diff
@@ -24,36 +24,62 @@
...
 
 
 class ForeignKeyRelationship(object):
   """Helps define a foreign key relationship between two models."""
 
   def __init__(self,
                referenced_table_name: str,
-               columns: Mapping[str, str]):
+               columns: Mapping[str, str],
+               single: bool = False):
```
Comment (dgorelik) [inline spanner_orm/foreign_key_relationship.py:36]: Remove param here, not needed

**Comment by dgorelik** [comment]
@akefeli, can you please take another look? I ended up making a good amount of changes after your first approval.

**Review by akefeli — APPROVED** [review]

**Pull request merged**